### PR TITLE
use STL random for random number generation

### PR DESCRIPTION
**Pull request by kejxu**

### CODE REVIEW

**Comment by ipa-mdl** [comment]
```
In file included from /root/catkin_ws/src/control_toolbox/src/dither.cpp:37:0:
/root/catkin_ws/src/control_toolbox/include/control_toolbox/dither.h: In member function ‘bool control_toolbox::Dither::init(const double&)’:
/root/catkin_ws/src/control_toolbox/include/control_toolbox/dither.h:86:3: error: expected ‘;’ before ‘}’ token
   }
   ^
```
This is easy to fix, but the kinetic jobs will fail anyway.

**Comment by bmagyar** [comment]
I've branched for melodic.
 @kejxu please edit `.travis.yml` and remove kinetic on this PR.

**Comment by kejxu** [comment]
> ```
> In file included from /root/catkin_ws/src/control_toolbox/src/dither.cpp:37:0:
> /root/catkin_ws/src/control_toolbox/include/control_toolbox/dither.h: In member function ‘bool control_toolbox::Dither::init(const double&)’:
> /root/catkin_ws/src/control_toolbox/include/control_toolbox/dither.h:86:3: error: expected ‘;’ before ‘}’ token
>    }
>    ^
> ```

thanks for catching this problem! I will check the current CI result and make sure this change does not break the build.

File: src/dither.cpp [hunk 1]
```diff
@@ -57,10 +59,12 @@ double Dither::update()
 
   // Generates gaussian random noise using the polar method.
   double v1, v2, r;
+  // uniform distribution on the interval [0.0, 1.0]
+  std::uniform_real_distribution<double> distribution(0.0, std::nextafter(1.0, std::numeric_limits<double>::max()));
```
Comment (ipa-mdl) [inline src/dither.cpp]: Does it make sense to use [-1 ,1] right away?

Comment (kejxu) [reply src/dither.cpp]: good point, change to [-1, 1]

**Comment by kejxu** [comment]
@ipa-mdl, just a friendly ping =) do we have any update for this pull request or https://github.com/ros-controls/control_toolbox/pull/74?

File: include/control_toolbox/dither.h [hunk 2]
```diff
@@ -71,6 +72,19 @@ class Dither
...
   double update();
 
+   /*
+   *\brief Dither gets an amplitude, must be >0 to initialize.
+   *
+   * Initialize random number generator with random_device for non-deterministic random numbers
+   *
+   *\param amplitude Amplitude of white noise output
+   */
+  bool init(const double &amplitude)
```
Comment (ipa-mdl) [inline include/control_toolbox/dither.h]: I'd prefer a static function to generate a non-deteministic seed.
This way user code would stay the same for use is test and production.

Comment (ipa-mdl) [reply include/control_toolbox/dither.h]:
```c++
Dither dither;
double seed = Dither::generateRandomSeed();
ROS_INFO_STREAM("Use seed " << seed);
dither.init(1.0, seed);
```

Comment (kejxu) [reply include/control_toolbox/dither.h]: > ```c++
> Dither dither;
> double seed = Dither::generateRandomSeed();
> ROS_INFO_STREAM("Use seed " << seed);
> dither.init(1.0, seed);
> ```

taking this as an example, sent out updated version

File: .travis.yml [hunk 3]
```diff
@@ -16,8 +16,6 @@ env:
   global:
     - ROS_PARALLEL_TEST_JOBS=-j1
   matrix:
-    - ROS_DISTRO=kinetic  ROS_REPO=ros              UPSTREAM_WORKSPACE=https://raw.github.com/ros-controls/ros_control/$ROS_DISTRO-devel/ros_control.rosinstall
```
Comment (ipa-mdl) [inline .travis.yml]: Please move this a separate,first commit. Thanks!

**Comment by bmagyar** [comment]
I'm happy with more C++11 making it to `ros_control` :)

File: .travis.yml (hunk 3 above)
Comment (kejxu) [reply .travis.yml]: sorry for the delay, Monday was a holiday here. moved `.travis.yml` change to https://github.com/ros-controls/control_toolbox/pull/76

**Comment by ipa-mdl** [comment]
@kejxu: Why did you revert this? I meant to rebase the branch to split one commit into two..

**Comment by kejxu** [comment]
> @kejxu: Why did you revert this? I meant to rebase the branch to split one commit into two..

ah, I totally misunderstood that. closing the other pull request right now

**Comment by kejxu** [comment]
> Thanks!
> 
> I think we should wait for Travis to pass the test again ([ros-infrastructure/rosdep#663](https://github.com/ros-infrastructure/rosdep/pull/663)).
> This ensures that the rebase did not break anything.

Thanks for the fix in `rosdep`. Pretty sure the build is going to pass, but I'm totally fine with waiting for a passing build =)

the travis error log shows a connection error as
```
ERROR: unable to process source [https://raw.githubusercontent.com/ros/rosdistro/master/releases/fuerte.yaml]:
	Failed to download target platform data for gbpdistro:
	HTTP Error 503: Backend unavailable, connection timeout
```
since the change in your other pull request is just changing `github` to `githubusercontent`, I think the tests are failing on this change due to temporary network issues. could you also re-queue builds on this change from time to time?

**Comment by ipa-mdl** [comment]
> Pretty sure the build is going to pass

I am sure as well :)

> since the change in your other pull request is changing `github` to `githubusercontent`, I think the tests are failing on this change due to temporary network issues

AFAIK raw.github.com is not supported anymore (?).

> could you also re-queue builds on this change from time to time?

I will restart the build as soon as the rosdep problem is resolved!

**Comment by kejxu** [comment]
looks like the build is passing, could we get a merge for this change? thanks =)

**Comment by bmagyar** [comment]
Thanks for the contribution!

**Pull request merged**